src/hotspot/os_cpu/bsd_x86/os_bsd_x86.cpp

```diff
@@ -472,7 +472,7 @@ bool PosixSignals::pd_hotspot_signal_handler(int sig, siginfo_t* info,
 
 #ifdef AMD64
       if (sig == SIGFPE  &&
-          (info->si_code == FPE_INTDIV || info->si_code == FPE_FLTDIV)) {
+          (info->si_code == FPE_INTDIV || info->si_code == FPE_FLTDIV MACOS_ONLY(|| info->si_code == FPE_FLTINV))) {
```

Comment:
Can you add a comment stating that MacOs (or Rosetta?) misreports FLTDIV as FLTINV?

Reply:
Will do.

Reply:
nit - can you put a period at the end of L475.